src/coreclr/jit/optimizer.cpp

```diff
@@ -4554,24 +4564,39 @@ PhaseStatus Compiler::optUnrollLoops()
 
             // Now fix up the exterior flow and pred list entries.
             //
-            // Control will fall through from HEAD to its successor, which is either
-            // the now empty TOP (if totalIter == 0) or the first cloned top.
+            // Control will fall through from the initBlock to its successor, which is either
+            // the pre-header HEAD (if it exists), or the now empty TOP (if totalIter == 0),
+            // or the first cloned top.
+            //
+            // If the initBlock is a BBJ_COND drop the condition (and make initBlock a BBJ_NONE block).
             //
-            // If the HEAD is a BBJ_COND drop the condition (and make HEAD a BBJ_NONE block).
+            // Also, make sure HEAD is a BBJ_NONE block.
             //
-            if (head->bbJumpKind == BBJ_COND)
+            if (initBlock->bbJumpKind == BBJ_COND)
```

Comment:
Could you explain the change here? From the definition of `lpInitBlock` it looks like this can be `head` or a `preheader` So we should be always using `initBlock` here instead of `head` right? 

```c++
        // Set to the block where we found the initialization for LPFLG_CONST_INIT loops.
        // Initially, this will be 'head', but 'head' might change if we insert a loop pre-header block.
        BasicBlock* lpInitBlock;
```

Reply:
Yeah, I removed the obsolete reference to `head` in the comment.